### PR TITLE
chore: Hot reload explains a skipped edit the same way regardless of which part detected it

### PR DESCRIPTION
**Pull request by hatayama**
## Summary
- Hot reload words every skip reason and introduced-type diagnostic in one place, so the sentence a user reads can no longer differ depending on which process produced it.
- Nothing hot reload prints changes: every response string stays byte-identical, verified against captured baselines from a real project run.

## User Impact
- Before, the same situation could be worded twice: the external transform worker composed English sentences and the Editor composed its own, and the two drifted apart as either side changed. A comment was the only thing tying the worker's payload fields to the Editor's, so a field added on one side alone reached users as a silently empty value.
- After, the worker reports a reason as a code plus its arguments, and one Editor renderer turns that into English. A wording change now happens once, and the worker and Editor payload shapes are compared field by field by a test, which names the type, the field, and the side it is missing from when they disagree.
- The reference doc now says where the wording lives, so the sentences are no longer searched for in the worker.

## Changes
- Replaced the worker's English skip-reason and diagnostic strings with a reason code plus arguments (and an optional free-form detail); `HotReloadWorkerReasonText` is the only place that composes the English a caller sees.
- Added a test that reads the built worker assembly's metadata and asserts, for all 14 payload type pairs, that the field names match and that every matching field describes the same kind of value. The camel-case approximation it relies on fails the test rather than being assumed.
- Removed the `// Keep in sync` comments the test replaces; the ones that point at a constant, a key format, or a hash format stay, since no test covers those.
- Split the production files this work touched that had grown past the length target, as pure moves.
- Documented the reason-code flow in the hot reload reference.

## Verification
- Real-project smoke: the plan's two hot reload runs (unsupported-member edit, newly introduced type, generic method) re-executed on this branch head. The full command output compares **0 diff lines** against the baselines captured before the refactor for all three captures. After `hot-reload --revert-all`, restoring the edited file and recompiling, `git status --porcelain` is empty.
- `run-tests --filter-type class` per class, single-flight, 33 hot reload test classes: all `Passed`, 0 failures — including `HotReloadOrchestratorTests` 155/155, `HotReloadWorkerReasonTextTests` 117/117, `TransformWorkerClientTests` 81/81, `TransformWorkerAddedFieldTests` 66/66, `TransformWorkerAddedMemberTests` 57/57, `TransformWorkerAddedPropertyTests` 40/40, `TransformWorkerIntroducedTypeTests` 33/33, `HotReloadIntroducedTypeFingerprintTests` 27/27, `HotReloadCrossFileE2ETests` 19/19, and the new wire-format test 2/2.
- `CODE_FILE_LENGTH_FAIL_ON_EXCEEDED=true scripts/check-file-length.sh`: exit 0, no file over the 500 SLOC limit.
- At the stricter 400 SLOC target used for this work, no file it touched is reported; the two remaining hot reload entries (433 and 435 SLOC) are files this work never changed.
- Byte-identical output is also pinned by the existing tests, whose expected-value literals are unchanged in this diff.

<a href="https://cubic.dev/pr/hatayama/unity-cli-loop/pull/2790?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: Packages/src/Editor/FirstPartyTools/HotReload/TransformWorker~/AddedFieldValueTypeWriteScan.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    {
+        ExpressionSyntax current = expression;
+        while (current != null)
+        {
+            if (AddedFieldBodyScan.IsStoreAddedValueTypeField(semanticModel, current, addedFieldCatalog))
+            {
+                return true;
+            }
+
+            current = TryGetReceiver(current);
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/FirstPartyTools/HotReload/TransformWorker~/AddedFieldValueTypeWriteScan.cs:94]: Out of scope for this pull request, and the behavior predates it. This scan is a pure move: the same `ReachesValueTypeAddedFieldRoot` / `TryGetReceiver` walk exists unchanged on the base branch in the file it was split out of, so nothing here changes what the receiver walk decides. This work is behavior-preserving by design and is verified against byte-identical baselines, so changing the scan result would break that property.

On the finding itself: the direction of the error is conservative. Returning `true` reports the edited method as skipped, which asks the user to compile rather than silently patching a body whose write might be lost. A narrower walk risks the opposite, which is the failure mode worth avoiding. Tightening it is a behavior change that belongs in its own change with its own test for the reference-type boundary case.
